pandas/tests/arithmetic/test_datetime64.py

```diff
@@ -2344,29 +2344,29 @@ def test_ufunc_coercions(self):
             assert result.freq == "2D"
 
         exp = date_range("2010-12-31", periods=3, freq="2D", name="x")
+
         for result in [idx - delta, np.subtract(idx, delta)]:
             assert isinstance(result, DatetimeIndex)
             tm.assert_index_equal(result, exp)
             assert result.freq == "2D"
 
+        # When adding/subtracting an ndarray (which has no .freq), the result
+        #  does not infer freq
```

Comment:
i think u need to pass freq=None when constructing th index rather than using a private method

Reply:
yah, first we need to fix the DTI constructor so that if you explicitly pass freq=None it wont override the arg's freq, ATM we get:

```
>>> dti = pd.date_range("2016-01-01", periods=3)
>>> pd.DatetimeIndex(dti, freq=None).freq
<Day>
```

Reply:
yeah i guess freq should be a no_default arg s can handle None

Reply:
yah.  ill make a PR for that in the AM

Reply:
working on the `freq=no_default` branch and its hairier than anticipated.  if we can push this through in the interim that'll help me keep momentum on the "check freq in assert_index_equal" project

Reply:
ok sure, yeah like to avoid using `_with_freq` in tests at all